packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -121,29 +123,28 @@ class BlockListBlock extends Component {
 
 	getBlockWidth( { nativeEvent } ) {
 		const { layout } = nativeEvent;
-		const { marginHorizontal, blockWidth } = this.props;
+		const { blockWidth } = this.state;
 
 		if ( ! blockWidth ) {
 			return;
 		}
 
-		const spacing = 2 * marginHorizontal;
 		const layoutWidth = Math.floor( layout.width );
 
-		if ( blockWidth - spacing !== layoutWidth ) {
-			if ( this.state.blockWidth !== layoutWidth ) {
-				this.setState( { blockWidth: layoutWidth } );
-			}
+		if ( blockWidth !== layoutWidth ) {
+			this.setState( { blockWidth: layoutWidth } );
 		}
 	}
 
 	getBlockForType() {
+		const { blockWidth } = this.state;
 		return (
 			<BlockForType
 				{ ...this.props }
 				onBlockFocus={ this.onFocus }
 				insertBlocksAfter={ this.insertBlocksAfter }
 				getBlockWidth={ this.getBlockWidth }
+				blockWidth={ blockWidth }
```

Comment:
I think that if we use the `blockWidth` only here there is no need for destructuring it :)